app/models/theme_site_setting.rb

```diff
@@ -99,6 +99,14 @@ def self.generate_theme_map
     theme_site_setting_values_map
   end
 
+  def self.generate_defaults_map
+    SiteSetting
+      .themeable_site_settings
+      .each_with_object({}) do |setting_name, map|
+        map[setting_name] = SiteSetting.defaults[setting_name]
+      end
```

Comment:
Can technically use `#to_h` with a block to turn an array into a hash. I think the `#to_h` itself signals the intent a bit better, but it's arguable whether it's more readable or not:

```suggestion
    SiteSetting
      .themeable_site_settings
      .to_h do |setting_name|
        [setting_name, SiteSetting.defaults[setting_name]]
      end
```